services/order.py

```diff
@@ -0,0 +1,43 @@
+from django.db import transaction
+from django.db.models import QuerySet
+from db.models import Ticket, Order, User, MovieSession
+
+
+def create_order(tickets: list[dict],
+                 username: str,
+                 date: str = None) -> QuerySet:
+
+    with transaction.atomic():
+
+        user = User.objects.get(username=username)
+        order = Order.objects.create(user=user)
+        tickets_instances = []
+        if date:
+            order.created_at = date
+            order.save()
```

Comment:
The checklist advises not to repeat yourself. You can call the `save()` method only once after the loop, instead of calling it conditionally here.